Automattic/newspack-plugin
# fix(ia): hide notifications

### All Submissions:






### Changes proposed in this Pull Request:

The current "hide notifications" code needed some fixes to work for the current wizards and new wizards too.  

This new solution moves `remove_notifications()` from the newspack class to the wizard abstract class.

To get an idea of what notifications are being removed, please see [this PR](https://github.com/Automattic/newspack-plugin/pull/3531) that was leaving the notices in place, but just adding margin.  We've now since decided/fixed that we'll actually remove the notices instead.

Another fix/change that was made was to move the rendering hook for the "admin-header-only" wizards to `in_admin_header` instead of `all_admin_notices` because we were removing the action that was also being used for rendering.  Because of this hook change, the location in the HTML where the admin-header was being rendered changed, so we needed to adjust the CSS based on this new HTML location.  Mainly this was needed to fix the mobile layout.

![image](https://github.com/user-attachments/assets/1e2746b9-d63f-49c2-b086-852a8e7fa2d9)

Finally, the CSS that originally added margins to the notifications (instead of removing notifications) was allowed to stay in the CSS (with some refactoring).  This is kind of a "fall-back" in case the PHP doesn't hide a notice for some reason it will still look OK on the page.

## Testing:

When testing, no notices should appear on any wizards (except WordPress core notices like "1 post moved to trash" and "saved").  

For Wizards that are "admin-header-only" on CPT pages, notices such as "post moved to trash" and other "bulk/edit" actions will still show.  These are internal to WordPress and provide helpful feedback to users/admins.

![image](https://github.com/user-attachments/assets/84dd23e4-13b5-484b-bf0f-a1487801379b)

Settings pages such as Network > Site Role (`admin.php?page=newspack-network`) will show a "saved" notice like the attached since these are rendered by core and are not removed.

![image](https://github.com/user-attachments/assets/c730adfc-fc2c-4b79-ad42-4b95696db29b)


#### How to test the changes in this Pull Request:

1. Pull this branch.
2. `npm run build`
3. Click on a variety of Wizards pages in wp-admin and verify no notices are shows.

### Other information:
